src/bp-search/classes/class-bp-search-bbpress-forums-replies.php

```diff
@@ -83,15 +91,7 @@ function ( $group_id ) {
 
 				$in = implode( '', $in );
 
-				$group_query = ' pm.meta_value IN ( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key = \'_bbp_group_ids\' AND meta_value IN(' . trim( $in, ',' ) . ') ) AND ';
-			}
-
-			if ( current_user_can( 'read_hidden_forums' ) ) {
-				$post_status = array( "'publish'", "'private'", "'hidden'" );
-			} elseif ( current_user_can( 'read_private_forums' ) ) {
-				$post_status = array( "'publish'", "'private'" );
-			} else {
-				$post_status = array( "'publish'" );
+				$group_query = ' ( pm.meta_value IN ( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key = \'_bbp_group_ids\' AND meta_value IN(' . trim( $in, ',' ) . ') ) OR pm.meta_value IN( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key != \'_bbp_group_ids\' and post_id in ( SELECT ID FROM ' . $wpdb->posts . ' WHERE post_type = \'forum\' AND post_status IN (' . join( ',', $post_status ) . ') ) ) ) AND ';
```

Comment:
@mahedihasannoman Why we need to inner Query 
I think simplest solution is just add `post status` with query

Reply:
Hi @dipskakadiya, Done as we discussed.